backend/masterScript_createMapLayers.py

```diff
@@ -0,0 +1,78 @@
+# Import data sources information
+from data.dataSources import *
+
+# Import  functions to call
+from assimilator import assimilate
+from generateLayer import generateLayer
+
+# Initialise counts to count successful and unsuccessful operations
+count_data = 0
+count_dataEnabled = 0
+count_dataSuccess = 0
+
+# Initialise list of data sources marked as disabled
+skipped_data = []
+
+
+'''
+Intended purpose of this script: 
+1. Feed data source metadata into assimilate()
+2. Generate geojson files from raw files 
+
+Assimilate () function:
+If input is csv then rows are inserted as properties into geojson boundary file
+else If input is geojson then point features are formatted into geojson
+
+generateLayer() function:
+Generates an objetice JS formatted entry in the layer.js from each geoJSON
+'''
+
+# Use source information to access filenames and metadata to produce geojsons 
+for data in DATASOURCES:
+
+    # Proceed with data unless marked as 'disabled'
+    if data["enabled"] == True:
+
+        # If data marked as 'geojson' then skip, or else recompile as geojson
+        if data["type"] == "geojson":
+            continue
+            
+        # Run Assimilate() for each data source
+        else:
+            geo = BOUNDARYFILES[data["res"]]
+            assimilate(
+                data["type"],
+                data["path"],
+                data["ID_name"],
+                geo["path"],
+                geo["ID_name"],
+                "dashboard/data/{}.geojson".format(data["name"]),
+            )
```

Comment:
This function invocation calls for better **encapsulation** -==> pls consider changing the `assimilate` synopsis to accept an object encapsulating this information.

In fact, if you think of it more carefully, this also matches the typing definition idea of Data Source I introduced earlier in a previous comment. 

Here you could've passed just a `Dataset` to `assimilate`? ;)
Looks like a plan.. ;D

Reply:
Let's discuss this too- it works for now and we're probably removing it anyway so I'd rather not spend time on non-essential fixes to temporary components